packages/legacy-client/src/client/convertWireToOsdkObject.ts

```diff
@@ -55,22 +66,22 @@ export function convertWireToOsdkObject<
   T extends keyof O["objects"] & string,
   O extends OntologyDefinition<any>,
 >(
+  context: ClientContext,
   obj: OsdkLegacyPropertiesFrom<O, T> & OntologyObject<T>,
-  ontologyDefinition: O,
 ): OsdkLegacyObjectFrom<O, T> {
-  const ontologyCache = cache.get(ontologyDefinition.metadata.ontologyRid);
+  const ontologyCache = cache.get(context.ontology.metadata.ontologyRid);
```

Comment:
I'm considering that the first level of cache might be able to just be folded into `ClientContext`